src/detext/run_detext.py

```diff
@@ -132,100 +136,117 @@ class Args(NamedTuple):
     _tokenization = {'choices': ['plain', 'punct']}
 
     resume_training: bool = False  # Whether to resume training from checkpoint in out_dir.
-    metadata_path: str = None  # The metadata_path for converted avro2tf avro data.
+    metadata_path: Optional[str] = None  # The metadata_path for converted avro2tf avro data.
 
     # tf-ranking related
     use_tfr_loss: bool = False  # whether to use tf-ranking loss.
     tfr_loss_fn: str = tfr.losses.RankingLossKey.SOFTMAX_LOSS  # tf-ranking loss
-    _tfr_loss_fn = {'choices': [tfr.losses.RankingLossKey.SOFTMAX_LOSS, tfr.losses.RankingLossKey.PAIRWISE_LOGISTIC_LOSS]}
-    tfr_lambda_weights: str = None  #
+    _tfr_loss_fn = {
+        'choices': [tfr.losses.RankingLossKey.SOFTMAX_LOSS, tfr.losses.RankingLossKey.PAIRWISE_LOGISTIC_LOSS]}
+    tfr_lambda_weights: Optional[str] = None  #
 
     use_horovod: bool = False  # whether to use horovod for sync distributed training
-
+    hvd_info: Optional[Dict[str, int]] = None
     # multitask training related
-    task_ids: List[int] = None  # All types of task IDs for multitask training. E.g., 1,2,3
+    task_ids: Optional[List[int]] = None  # All types of task IDs for multitask training. E.g., 1,2,3
     _task_ids = __comma_split_list(int)
-    task_weights: List[float] = None  # Weights for each task specified in task_ids. E.g., 0.5,0.3,0.2
+    task_weights: Optional[List[float]] = None  # Weights for each task specified in task_ids. E.g., 0.5,0.3,0.2
     _task_weights = __comma_split_list(float)
 
-
-def get_hparams():
+    def __late_init__(self):
```

Comment:
This seems to be specific to smart-args. Can you add some comments to the function wrt what this is doing?

Reply:
`__late_init__` will be called after an argument was created. It's used for post-processing, exactly for the purpose of the refactored code. will add some comments.